#!/usr/bin/env python
PACKAGE = "tuw_iwos_odometer"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

odometer_enum = gen.enum([gen.const("odometer_motor", int_t, 0, "motor sensor based odometer"),
                          gen.const("odometer_sensor", int_t, 1, "motor sensor and IMU based odometer")],
                         "odometer type")

gen.add("odometer", int_t, 0, "odometer type", 0, 0, 1, edit_method=odometer_enum)

gen.add("publish_odom_message", bool_t, 0, "publish odom message", True)
gen.add("broadcast_odom_transform", bool_t, 0, "broadcast odom transform", True)

gen.add("linear_velocity_tolerance", double_t, 0, "tolerance for linear velocity matching", 0.025, 0.0, 1.0)
gen.add("angular_velocity_tolerance", double_t, 0, "tolerance for angular velocity matching", 0.025, 0.0, 1.0)
gen.add("steering_position_tolerance", double_t, 0, "tolerance for steering position matching", 0.025, 0.0, 1.0)

gen.add("calculation_iteration_duration", double_t, 0, "number of odometer calculation iterations", 0.001, 0, 0.1)

exit(gen.generate(PACKAGE, "tuw_iwos_odometer", "OdometerNode"))
